znnl/data/decision_boundary.py

```diff
@@ -0,0 +1,193 @@
+"""
+ZnNL: A Zincwarecode package.
+
+License
+-------
+This program and the accompanying materials are made available under the terms
+of the Eclipse Public License v2.0 which accompanies this distribution, and is
+available at https://www.eclipse.org/legal/epl-v20.html
+
+SPDX-License-Identifier: EPL-2.0
+
+Copyright Contributors to the Zincwarecode Project.
+
+Contact Information
+-------------------
+email: zincwarecode@gmail.com
+github: https://github.com/zincware
+web: https://zincwarecode.com/
+
+Citation
+--------
+If you use this module please cite us with:
+
+Summary
+-------
+Data generator for decision boundary problems.
+"""
+import jax
+import jax.numpy as np
+import matplotlib.pyplot as plt
+import numpy as onp
+
+from znnl.data.data_generator import DataGenerator
+from znnl.utils.prng import PRNGKey
+
+
+def linear_boundary(data: onp.ndarray, gradient: float, intercept: float) -> np.ndarray:
```

Comment:
This seems to be made for 2d data. 
Would it be more generic, to define a vector of n-dims instead of gradient along which the decision is made?

Reply:
I guess you could do this for straight lines, but quickly becomes intractable for this case. We can think about it maybe for n dimensions.

Reply:
I thought you placed the two functions outside the class to generically make use of them. 
One could define a normal vector to the plane of separation, which would be generic for n-d. 
What do you think about that?

Reply:
Maybe we can discuss it in person

Reply:
Opened #103 to address this in the future.